### PR TITLE
Allow to install Presto RPM with Java 11

### PR DESCRIPTION
**Pull request by kokosing**
Allow to install Presto RPM with Java 11

### CODE REVIEW

**Comment by electrum** [comment]
With https://github.com/prestosql/presto/pull/2064 we could easily test this using a different Docker image that contains Java 11

File: presto-server-rpm/src/main/rpm/preinstall [hunk 1]
```diff
@@ -22,7 +22,11 @@ check_if_correct_java_version() {
 # candidate for JAVA_HOME).
   JAVA_VERSION=$(java_version "$1")
   JAVA_UPDATE=$(echo $JAVA_VERSION | cut -d'_' -f2)
-  if [[ ("$JAVA_VERSION" > "1.8") && ($JAVA_UPDATE -ge 161) ]]; then
+  JAVA_MAJOR=$(echo $JAVA_VERSION | cut -d'.' -f1)
+  if [[ ("$JAVA_MAJOR" -ge "11") ]]; then
```
Comment (sopel39) [inline presto-server-rpm/src/main/rpm/preinstall]: what about other versions? Can we make the mechanism generic?

File: presto-server-rpm/src/main/rpm/preinstall [hunk 2]
```diff
@@ -22,7 +22,11 @@ check_if_correct_java_version() {
...
+    echo "JAVA11_HOME=$1" > /tmp/presto_env.sh
+    return 0
+  elif [[ ("$JAVA_VERSION" > "1.8") && ($JAVA_UPDATE -ge 161) ]]; then
     echo "JAVA8_HOME=$1" > /tmp/presto_env.sh
```
Comment (sopel39) [inline presto-server-rpm/src/main/rpm/preinstall]: shouldn't we make this generic instead?

**Comment by electrum** [comment]
#2064 is merged, so please try adding a test for this in `io.prestosql.server.rpm.ServerIT`

File: presto-server-rpm/src/main/rpm/preinstall (hunk 1 above)
Comment (kokosing) [reply presto-server-rpm/src/main/rpm/preinstall]: This is quite generic for example `12 > 11`. However then env variable `JAVA11_HOME` has unfortunate name.

**Comment by kokosing** [comment]
Locally:
```
[INFO] Running io.prestosql.server.rpm.ServerIT
[main] INFO org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
[main] INFO org.testcontainers.dockerclient.UnixSocketClientProviderStrategy - Accessing docker with local Unix socket
[main] INFO org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
[main] INFO org.testcontainers.DockerClientFactory - Docker host IP address is localhost
[main] INFO org.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 19.03.1
  API Version: 1.40
  Operating System: Docker Desktop
  Total Memory: 6209 MB
[main] INFO org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for quay.io
[main] INFO org.testcontainers.DockerClientFactory - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
        ℹ︎ Checking the system...
        ✔ Docker version should be at least 1.6.0
        ✔ Docker environment should have more than 2GB free disk space
[main] INFO 🐳 [prestodev/centos7-oj11:latest] - Creating container for image: prestodev/centos7-oj11:latest
[main] INFO org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for index.docker.io
[main] INFO org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for index.docker.io
[main] INFO 🐳 [prestodev/centos7-oj11:latest] - Starting container with ID: 50b196405184860a9f0a2fbde0cf4815a3adfa3c3c67948c7f3148967d5deee2
[main] INFO 🐳 [prestodev/centos7-oj11:latest] - Container prestodev/centos7-oj11:latest is starting: 50b196405184860a9f0a2fbde0cf4815a3adfa3c3c67948c7f3148967d5deee2
[main] INFO 🐳 [prestodev/centos7-oj11:latest] - Container prestodev/centos7-oj11:latest started in PT31.856S
[main] INFO 🐳 [prestodev/centos7-oj8:latest] - Creating container for image: prestodev/centos7-oj8:latest
[main] INFO org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for index.docker.io
[main] INFO 🐳 [prestodev/centos7-oj8:latest] - Starting container with ID: 6055124ed03ed31bbe081c4162104922100902fb2a048701fbc3cd1c86e36547
[main] INFO 🐳 [prestodev/centos7-oj8:latest] - Container prestodev/centos7-oj8:latest is starting: 6055124ed03ed31bbe081c4162104922100902fb2a048701fbc3cd1c86e36547
[main] INFO 🐳 [prestodev/centos7-oj8:latest] - Container prestodev/centos7-oj8:latest started in PT30.612S
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 74.033 s - in io.prestosql.server.rpm.ServerIT
```

**Comment by kokosing** [comment]
Depends on https://github.com/prestosql/docker-images/pull/41

**Comment by kokosing** [comment]
@electrum @sopel39 Comments addressed. Please have another look.

File: presto-server-rpm/src/main/rpm/preinstall [hunk 3]
```diff
@@ -40,6 +44,7 @@ if ! check_if_correct_java_version "$JAVA8_HOME" && ! check_if_correct_java_vers
       /usr/java/jdk1.8* \
       /usr/java/jre1.8* \
       /usr/jdk64/jdk1.8* \
+      /usr/lib/jvm/java-11-* \
```
Comment (electrum) [inline presto-server-rpm/src/main/rpm/preinstall]: Should we put this first in the list so that it's found first?

**Pull request merged**